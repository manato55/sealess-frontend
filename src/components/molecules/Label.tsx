import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Link from 'next/link';
import { useRouter } from 'next/router';

interface Props {
  path: string;
  tagName: string;
  pathMatch?: string;
}

export const Label = (props: Props): React.ReactElement => {
  const router = useRouter();
  const [currentPathname, setCurrentPathname] = useState<string>('');

  useEffect(() => {
    setCurrentPathname(router.pathname);
  }, [router.pathname]);

  return (
    <>
      <Link href={props.path}>
        <a>
          {currentPathname.match(props.pathMatch || `${props.path}*`) ? (
            <CurrentTab>{props.tagName}</CurrentTab>
          ) : (
            <p>{props.tagName}</p>
          )}
        </a>
      </Link>
    </>
  );
};

const CurrentTab = styled.p`
  background: #1e90ff;
`;

export default React.memo(Label);
