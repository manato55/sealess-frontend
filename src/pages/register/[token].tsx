import React, { useEffect, useState, useCallback } from 'react';
import { useRouter } from 'next/router';
import { useAuthenticate } from '../../hooks/useAuthenticate';
import Password from '../../components/molecules/Password';
import Button from '../../components/atoms/Button';
import AuthWrapper from '../../components/atoms/AuthWrapper';

export const Token = (): React.ReactElement => {
  const router = useRouter();
  const { tokenCheck } = useAuthenticate();
  const { officialRegistryForOrdinaryUser } = useAuthenticate();
  const [paramsToken, setParamsToken] = useState<any>(router.query.token);
  const [tokenChecker, setTokenChecker] = useState<boolean>(false);
  const [password, setPassword] = useState<string>('');

  useEffect(() => {
    const initialAction = async () => {
      const res = await tokenCheck(paramsToken);
      if (res) {
        setTokenChecker(true);
      }
    };
    initialAction();
  }, [paramsToken]);

  async function submitRegister(): Promise<void> {
    const data: {
      token: string;
      password: string;
    } = {
      token: paramsToken,
      password: password,
    };
    await officialRegistryForOrdinaryUser(data);
  }

  return (
    <>
      {tokenChecker ? (
        <AuthWrapper>
          <Password password={password} setPassword={setPassword} />
          <Button onClick={() => submitRegister()}>登録</Button>
        </AuthWrapper>
      ) : (
        'Loading...'
      )}
    </>
  );
};

export default Token;
