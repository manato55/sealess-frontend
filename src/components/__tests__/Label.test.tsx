import React from 'react';
import Label from '../molecules/Label';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';
import 'jsdom-global/register';
import { renderHook, act } from '@testing-library/react-hooks';
import renderer from 'react-test-renderer';
import { withHooks } from 'jest-react-hooks-shallow';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      pathname: '/history',
    };
  },
}));

describe('Label test', () => {
  test('snapshot', () => {
    const tree = renderer.create(<Label path={'test'} tagName={'foo'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
