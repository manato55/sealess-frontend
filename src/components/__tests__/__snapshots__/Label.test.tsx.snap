// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label test snapshot 1`] = `
<a
  href="test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <p>
    foo
  </p>
</a>
`;
